tests/e2e/test_api_xai_sanity.py

```diff
@@ -114,38 +115,43 @@ def test_inference_xai(self, multilabel, hierarchical):
-            # # Save OV IR model
-            # task._model_ckpt = osp.join(temp_dir, "weights.pth")
-            # exported_model = ModelEntity(None, task_environment.get_model_configuration())
-            # task.export(ExportType.OPENVINO, exported_model)
-            # os.makedirs(temp_dir, exist_ok=True)
-            # save_model_data(exported_model, temp_dir)
-            #
-            # # Infer OV IR model
-            # load_weights_ov = osp.join(temp_dir, "openvino.xml")
-            # task_environment.model = read_model(task_environment.get_model_configuration(), load_weights_ov, None)
-            # task = ClassificationOpenVINOTask(task_environment=task_environment)
-            # predicted_dataset_ov = task.infer(
-            #     dataset.with_empty_annotations(),
-            #     InferenceParameters(is_evaluation=False),
-            # )
-            #
-            # # Check saliency maps OV task
-            # saliency_maps_check(predicted_dataset_ov, task_labels, self.ref_raw_saliency_shapes[model_template.name])
+                # Save OV IR model
+                task._model_ckpt = osp.join(temp_dir, "weights.pth")
+                exported_model = ModelEntity(None, task_environment.get_model_configuration())
+                task.export(ExportType.OPENVINO, exported_model, dump_features=True)
+                os.makedirs(temp_dir, exist_ok=True)
+                save_model_data(exported_model, temp_dir)
+
+                # Infer OV IR model
+                load_weights_ov = osp.join(temp_dir, "openvino.xml")
+                task_environment.model = read_model(task_environment.get_model_configuration(), load_weights_ov, None)
+                task = ClassificationOpenVINOTask(task_environment=task_environment)
+                _, dataset = self.init_environment(hyper_parameters, model_template, multilabel, hierarchical, 20)
+                predicted_dataset_ov = task.infer(dataset.with_empty_annotations(), inference_parameters)
+
+                # Check saliency maps OV task
+                saliency_maps_check(
+                    predicted_dataset_ov,
+                    task_labels,
+                    self.ref_raw_saliency_shapes[model_template.name],
+                    processed_saliency_maps=processed_saliency_maps,
+                    only_predicted=only_predicted,
+                )
 
 
 class TestOVDetXAIAPI(DetectionTaskAPIBase):
     ref_raw_saliency_shapes = {
-        "ATSS": (4, 4),
+        "ATSS": (6, 8),
         "SSD": (13, 13),
         "YOLOX": (13, 13),
     }
 
     @e2e_pytest_api
     def test_inference_xai(self):
         with tempfile.TemporaryDirectory() as temp_dir:
-            hyper_parameters, model_template = self.setup_configurable_parameters(DEFAULT_DET_TEMPLATE_DIR, num_iters=2)
+            hyper_parameters, model_template = self.setup_configurable_parameters(
+                DEFAULT_DET_TEMPLATE_DIR, num_iters=100
```

Comment:
Do we need 100 liters for test code? I think it's quiet long even with e2e..

Reply:
Agree with @JihwanEom 's opinion. could we reduce this parameter?

Reply:
Sorry for not paying required attention to it. I reduced to 15. I tested, this is the minimal num of iters to get somewhat trained model. Is it ok for e2e?